.github/workflows/buildDeploy.yml

```
@@ -1,8 +1,79 @@
-# Placeholder to allow triggering of pipeline in branches
 name: Build and deploy docker image
 
-# Controls when the workflow will run
-on: [workflow_dispatch]   # Allows you to run this workflow manually from the Actions tab
-  # Triggers the workflow on push or pull request events but only for the "main" branch
-  # push:
+on: 
+
+  workflow_dispatch: # Allows manual workflow trigger
+  repository_dispatch: # Allows API workflow trigger
+
+  # Triggers the workflow on pull request events but only for the "main" branch
+  # pull_request:
   #   branches: [ "main" ]
+
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  AWS_REGION: eu-west-2
+  ECR_REPOSITORY: app-frontend
+  IMAGE_TAG: ${{ github.event.client_payload.COMMIT_HASH != null || github.sha }}
+  KUBE_CONFIG_DATA: ${{ secrets.KUBE_CONFIG_DATA }}
+  KUBE_NAMESPACE: default
+  DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+
+jobs:
+
+  buildDeploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Cancel Previous Runs               
+        uses: styfle/cancel-workflow-action@0.4.1
+        with:                                    
+          access_token: ${{ github.token }}
+
+      - name: Echo Env Vars through Context
+        run: |
+          echo "$GITHUB_CONTEXT"                 
+
+      - name: Configure AWS credentials                          
+        uses: aws-actions/configure-aws-credentials@v1           
+        with:                                                    
+          aws-access-key-id: ${{ env.AWS_ACCESS_KEY_ID }}        
+          aws-secret-access-key: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Login to Amazon ECR            
+        id: login-ecr                        
+        uses: aws-actions/amazon-ecr-login@v1
+      
+      - name: Checkout                                  
+        uses: actions/checkout@v2
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          context: ${{ github.workspace }}
+          file: ./Dockerfile
+          push: true
+          tags: 204031746016.dkr.ecr.eu-west-2.amazonaws.com/app-frontend:${{ env.IMAGE_TAG }}
```

Shouldn't the region and cluser id's _(I presume)_ be stored inside env variables / secretes ?

```suggestion
          tags: ${{ env.AWS_CLUSTER_ID }}.${{ env.AWS_REGION }}.amazonaws.com/app-frontend:${{ env.IMAGE_TAG }}
```

---

fair suggestion and I did consider it. the reason why I left it hardcoded is because the ECR repo where all the images will be based in will always be based in that account/region, as we will only have 1 image version that gets deployed into all envs, so they are not really variables.

---

Is that "cluster id" (or whatever it is), part of the private repo name ? Since honestly that's what triggered my alarm bells initially, not necessarily the region :)

---

That number is the AWS account id, which is the account id of the QA account where the images will be stored. The prod environment and any other environment will be using that account to pull images from. 

I'll create a variable for it, just to not have it public in the pipeline.